### PR TITLE
Add a legend and vary color by copy number

### PR DESCRIPTION
**Pull request by urbanslug**
Increase the darkness of the shade with increasing copy number
Fix resetting of color between increased copy number and inverted cells.
Add a color array to mobx store

output

Used this for the shades of #6a6a6a from https://www.color-hex.com/color/6a6a6a

image

### CODE REVIEW

**Comment by josiahseaman** [comment]
That's a seriously cool gif.  Feedback: The legend needs a second row which is the red color for inversion under increasing copy numbers.  We had talked about combining copy and inversion coloring rules.  This makes it more explicit: two color palettes, one neutral gray for + strand and one red tinted for - strand.  

This is great progress!  Keep up the good work.

P.S. How did you make that smooth gif?

File: src/ComponentConnectorRect.js [hunk 1]
```diff
@@ -84,15 +84,21 @@ export class MatrixCell extends React.Component {
...
 
-    if (this.props.item[0] > 1) {
-      color = "#6A6A6A";
+    if (copyNumber > 1 && !inverted) {
+      // 11 items is number of colors in copyNumberColorArray
+      if (copyNumber < 10) {
+        color = this.props.store.copyNumberColorArray[copyNumber];
+      } else {
+        color = this.props.store.copyNumberColorArray[10];
+      }
```
Comment (josiahseaman) [inline src/ComponentConnectorRect.js:97]: Your code looks really clean!  This is the point where you'd access a second palette array for invesion red colors.

**Review by josiahseaman — CHANGES_REQUESTED** [review]
What you have is a great improvement.  Let's do one more color palette and then we'll be able to get this to master before our public release.

**Comment by urbanslug** [comment]
What do you think about the legend blocking the matrix?

> The legend needs a second row which is the red color for inversion under increasing copy numbers. We had talked about combining copy and inversion coloring rules. 

What's the color would be best for cells which have increased copy number **and**  are inverted? Also do you mean a second **coloumn**?

What do you think about this?
image
or 
image

---

For the gif: 
There are probably easier ways to do it though like firefox add ons but I couldn't find one fast enough so I used quicktime to record the screen into a video (.mov) file then converted it to a gif using ffmpeg. 
I got the instructions for generating the gif here [superuser: How can I get ffmpeg to convert a .mov to a .gif?](https://superuser.com/a/436109).
 - I skipped the convert because it was small
 - I set resolution 1366x768 to have a visible gif

**Comment by urbanslug** [comment]
Something like this?

image

**Comment by josiahseaman** [comment]
Hi Njagi, This is a terrific improvement.  I like the two spectrums, especially the one going to black because it preserves a consistent meaning for darkening color = higher copy number.  I think the legend could be compacted much more, so that the color boxes take up the majority of the space.  Perhaps vertical text would help?  I also find color matching Text in a legend makes things clear.  You could make the actual text of "Inverted" red and allow it to overlap with the other column space.  The color connects these ideas in our mind.  You could also put "Inverted" at the bottom or along the side to indicate it's not competing, but complementing the idea of "Copy Number".

**Pull request merged**